src/fireedge/src/modules/components/Tabs/Showback/index.js

```diff
@@ -0,0 +1,293 @@
+    const [topChartsData, setTopChartsData] = useState([])
+    const [transformedResult, setTransformedResult] = useState()
+
+    // Create hooks for date range
+    const [dateRange, setDateRange] = useState({
+      startDate: DateTime.now().minus({ months: 1 }),
+      endDate: DateTime.now(),
+    })
+
+    const handleDateChange = (newDateRange) => {
+      setDateRange(newDateRange)
+    }
+
+    // Hook for fetch data
+    const [getShowback, queryData] = VmAPI.useLazyGetShowbackPoolFilteredQuery()
+
+    // Call the API to refetch data
+    const refetchData = () => {
+      const paramName = groups ? 'group' : 'user'
+      getShowback({
+        [paramName]: id,
+        startMonth: dateRange.startDate.month,
+        startYear: dateRange.startDate.year,
+        endMonth: dateRange.endDate.month,
+        endYear: dateRange.endDate.year,
+      })
+    }
+
+    // Refetch data when click on Get showback button
+    const handleGetShowbackClick = () => refetchData()
+
+    // First render of the component
+    useEffect(() => refetchData(), [])
+
+    // Hook after fetch data
+    useEffect(() => {
+      if (!queryData.isLoading && queryData.data) {
+        // Transform data
+        const transformedResultData = transformApiResponseToDataset(
+          queryData,
+          keyMap,
+          metricKeys,
+          labelingFunc
+        )
+
+        // Update data
+        setTransformedResult(transformedResultData)
+        setTopChartsData([transformedResultData].map(aggregateTotalCostByMonth))
+      }
+    }, [queryData])
+
+    // Create dataset function
+    const aggregateTotalCostByMonth = (datasetWrapper) => {
+      const dataset = datasetWrapper.dataset
+
+      if (!dataset.data || dataset.data.length === 0) {
+        return {
+          ...dataset,
+          isEmpty: datasetWrapper.isEmpty,
+        }
+      }
+
+      const aggregated = dataset.data.reduce((acc, record) => {
+        if (
+          record.MONTH &&
+          record.totalCost !== null &&
+          record.totalCost !== undefined
+        ) {
+          if (!acc[record.MONTH]) {
+            acc[record.MONTH] = { ...record, totalCost: 0 }
+          }
+          acc[record.MONTH].totalCost += parseFloat(record.totalCost)
+        }
+
+        return acc
+      }, {})
+
+      // Return dataset
+      return {
+        id: dataset.id,
+        data: Object.values(aggregated),
+        metrics: dataset.metrics,
+        label: dataset.label,
+        isEmpty: datasetWrapper.isEmpty,
+      }
+    }
+
+    // Show loading component
+    if (
+      queryData.isLoading ||
+      queryData.isFetching ||
+      queryData.isError ||
+      !queryData.data ||
+      !topChartsData ||
+      topChartsData.length === 0
+    ) {
+      return (
+        <LoadingDisplay
+          isLoading={queryData.isLoading || queryData.isFetching}
+        />
+      )
+    }
+
+    const topMetricNamesTranslated = mapValues(topMetricNames, (value, key) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).